src/main/flight/pos_hold.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * This file is part of Betaflight.
+ *
+ * Betaflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Betaflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Betaflight. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "pg/pos_hold.h"
+
+#ifdef USE_ALT_HOLD_MODE
+#include "common/time.h"
+#include "io/gps.h"
+
+#define POSHOLD_TASK_RATE_HZ 100 // hz
+
+typedef struct {
+    bool isPosHoldRequested;
```

Comment:
Make a function `bool posHoldIsActive()` instead

Reply:
I had enough difficulty getting the logic to work, as it is :-)
Simplified to just use the flight mode as the indicator that the pilot is requesting position hold.
In that sense the flight mode indicates that it is requested though it may not be achievable (eg no 3D fix)